07.ls_object/formatter.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require_relative 'file_detail'
+
+class Formatter
+  COL = 3
+
+  def initialize(names)
+    @names = names
+  end
+
+  def format_short
+    row_count = @names.size.ceildiv(COL)
+    max_bytesize = @names.max_by(&:bytesize).bytesize
+    cols = @names.each_slice(row_count)
+    lines = Array.new(row_count) do |idx|
+      cols.map do |col|
+        col[idx].ljust(max_bytesize)
+      end
+    end
+    lines.each do |name|
+      puts name.join(' ')
+    end
+  end
+
+  def format_long
+    file_details = @names.map { |name| FileDetail.new(name) }
+    puts "total #{file_details.map(&:blocks).sum}"
+    max_nlink = max_size(:nlink, file_details)
+    max_owner = max_size(:owner, file_details)
+    max_group = max_size(:group, file_details)
+    max_size = max_size(:size, file_details)
+    file_details.each do |file_detail|
+      name = file_detail.name
+      nlink = file_detail.nlink.to_s.rjust(max_nlink)
```

Comment:
```suggestion
    hoge = piyo(file_details)
    file_details.each do |file_detail|
      name = file_detail.name
      nlink = file_detail.nlink.to_s.rjust(hoge[:nlink])
```

各項目の最大長をハッシュで返すメソッドを定義すると、処理がスッキリしそうです

Reply:
修正しました！